client/coral-embed-stream/src/containers/Stream.js

```diff
@@ -112,7 +112,8 @@ class StreamContainer extends React.Component {
 
   componentDidMount() {
     if (this.props.previousTab) {
-      this.props.data.refetch();
+      this.props.data.refetch()
+        .then(() => this.props.setCommentCountCache(0));
```

Comment:
Great idea @gobengo, I think this should work, then we can also remove the logic from here: https://github.com/coralproject/talk/blob/0dfec6e619b622f767a01e17d6038a37ad8bd0c3/client/coral-embed-stream/src/containers/Embed.js#L40